presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -335,11 +335,21 @@ private boolean updateMemoryWithYieldInfo()
         }
         // Operator/driver will be blocked on memory after we call setBytes.
         // If memory is not available, once we return, this operator will be blocked until memory is available.
-        localUserMemoryContext.setBytes(memorySize);
+        updateMemory(memorySize);
         // If memory is not available, inform the caller that we cannot proceed for allocation.
         return operatorContext.isWaitingForMemory().isDone();
     }
 
+    private void updateMemory(long memorySize)
+    {
+        if (partial || useSystemMemoryPool) {
```

Comment:
Are both `partial` and `useSystemMemoryPool` needed? The field `partial` looks like it's used to decide whether we want to do user or system allocation. 

Also having multiple variables/fields (`partial`, `maxPartialMemory`, `useSystemMemoryPool`) for taking memory-related decisions make it non-trivial to reason about what's going on in this class.

Reply:
Yeah, that's a good comment. I think now, since we have the `useSystemMemory` flag, i can set this flag to `true` from the outside (`LocalExecutionPlanner`) in case of partial aggregation. That will simplify the tests as well.